packages/nocodb/src/services/app-hooks/app-hooks.service.ts

```diff
@@ -0,0 +1,300 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IEventEmitter } from '../../modules/event-emitter/event-emitter.interface';
+import type {
+  ApiCreatedEvent,
+  ApiTokenCreateEvent,
+  ApiTokenDeleteEvent,
+  AttachmentEvent,
+  BaseEvent,
+  FormColumnEvent,
+  GridColumnEvent,
+  MetaDiffEvent,
+  OrgUserInviteEvent,
+  PluginEvent,
+  PluginTestEvent,
+  ProjectUserResendInviteEvent,
+  ProjectUserUpdateEvent,
+  RelationEvent,
+  SharedBaseEvent,
+  SyncSourceEvent,
+  UIAclEvent,
+  UserEmailVerificationEvent,
+  UserPasswordChangeEvent,
+  UserPasswordForgotEvent,
+  UserPasswordResetEvent,
+  ViewColumnEvent,
+  WebhookEvent,
+} from './interfaces';
+import type { AppEvents } from 'nocodb-sdk';
+import type {
+  ColumnEvent,
+  FilterEvent,
+  ProjectCreateEvent,
+  ProjectDeleteEvent,
+  ProjectInviteEvent,
+  ProjectUpdateEvent,
+  SortEvent,
+  TableEvent,
+  UserSigninEvent,
+  UserSignupEvent,
+  ViewEvent,
+  WelcomeEvent,
+  WorkspaceEvent,
+  WorkspaceInviteEvent,
+} from './interfaces';
+
+const ALL_EVENTS = '__nc_all_events__';
+
+@Injectable()
+export class AppHooksService {
```

Comment:
Events for Map are missing? double check the items in `AppEvents`

P.S ignore if they belong to ViewEvent.